compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -1767,15 +1921,16 @@ class Objects(using Context @constructorOnly):
     if target == klass then
       thisV
     else if target.is(Flags.Package) then
-      Bottom
+      Package(target) // TODO: What is the semantics for package.this?
     else if target.isStaticObject then
       val res = ObjectRef(target.moduleClass.asClass)
       if elideObjectAccess then res
       else accessObject(target)
     else
       thisV match
         case Bottom => Bottom
-        case Cold => Cold
+        case UnknownValue => UnknownValue
```

Comment:
Can these ever happen? Can `thisV` be `Bottom`, `UnknownValue`, or `TopWidenedValue`?

Reply:
Theoretically `thisV` should always have type `Ref` (or `ThisValue`), but call sites of `resolveThis` do not have such enforcement. Shall we also refactor `resolveThis` in another PR?